lib/Echidna/Exec.hs

```diff
@@ -2,7 +2,10 @@
     LambdaCase, StrictData, TemplateHaskell #-}
 
 module Echidna.Exec (
-    checkETest
+    TestFunction
+  , checkETest
```

Comment:
can we make these names a bit more clear? I think in retrospect `checkETest` was already too ambiguous, and I think this compounds the problem. I'd also probably prefer to merge them into one function that takes a `TestFunction` as a parameter, but don't feel quite as strongly there.